table_engine/src/partition/rule/df_adapter/extractor.rs

```diff
@@ -67,6 +67,30 @@ impl FilterExtractor for KeyExtractor {
                     }
                     _ => None,
                 },
+                Expr::InList {
+                    expr: _,
+                    list,
+                    negated,
+                } => {
+                    // The 'not in' syntax is not currently supported.
+                    if negated {
+                        continue;
+                    }
```

Comment:
We can use tuple match to check here and above, for example:
`if let (Expr::Column(col), true) = (*expr, negated) `